hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java

```diff
@@ -197,4 +198,118 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     return omClientResponse;
   }
 
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
```

Comment:
Why do we need bucketLayout for keyDelete? We deduce the bucketLayout in OzoneManagerRatisUtils and create appropriate request right . Is this method only created for unit test?

Reply:
Yes, this method is necessary for the test. Also, we need `bucketLayout` as a parameter because we need to set the `bucketLayout` in the `OmBucketInfo`, earlier for FSO(File System Optimized) cases we were dependent on a flag `isBucketFSOptimized` but with the help of this PR, we will deduce the `bucketLayout` directly from `OmBucketInfo` hence, we need to set the appropriate `bucketLayout`. I hope this clears your doubt, if you have more questions please feel free to add them!